RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -48,61 +66,157 @@ namespace {
   inline int nPixelHits(reco::Track const & tk) {
     return tk.hitPattern().numberOfValidPixelHits();
   }
-      
+  
   inline float dz(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dz(bestVertex));
+    return std::abs(trk.dz(bestVertex));
   }
   inline float dr(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dxy(bestVertex));
+    return std::abs(trk.dxy(bestVertex));
+  }
+
+  inline void dzCut_par1(reco::Track const & trk, int & nLayers, const float * par, float dzCut[]) {
+    float dzE =  trk.dzError();
+    for (int i=2; i>=0; --i) {
+      dzCut[i] = powN(par[i]*nLayers,4)*dzE;
+    }
+  }
+  inline void drCut_par1(reco::Track const & trk, int & nLayers, const float* par, float drCut[]) {
+    float drE =  trk.d0Error();
+    for (int i=2; i>=0; --i) {
+      drCut[i] = powN(par[i]*nLayers,4)*drE;
+    }
   }
+  
+  inline void dzCut_par2(reco::Track const & trk, int & nLayers, const float * par, float dzCut[]) {
+    float pt = float(trk.pt());
+    float eta = float(trk.eta());
 
+    // parametrized d0 resolution for the track pt
+    float nomd0E = sqrt(0.003*0.003+(0.001/pt)*(0.001/pt));
+    // parametrized z0 resolution for the track pt and eta
+    float nomdzE = nomd0E*(std::cosh(eta));
```

Comment:
was there as written in the old implementation (sic)
Yes, need to be implemented as p/pt (or equivalent)
in nay case this will be used ONLY for pixel-doublet at HLT to make it backward compatible
I do not exclude we can find a retune that does not require this "funny" parameterization